app/v2/controllers/requestParsers/validators/SubmitForeignPropertyBsasValidator.scala

```diff
@@ -0,0 +1,113 @@
+        doValidationFor("/foreignProperty/expenses/repairsAndMaintenance", foreignProperty.expenses.flatMap(_.repairsAndMaintenance))(withAdjustmentValidations),
+        doValidationFor("/foreignProperty/expenses/financialCosts", foreignProperty.expenses.flatMap(_.financialCosts))(withAdjustmentValidations),
+        doValidationFor("/foreignProperty/expenses/professionalFees", foreignProperty.expenses.flatMap(_.professionalFees))(withAdjustmentValidations),
+        doValidationFor("/foreignProperty/expenses/travelCosts", foreignProperty.expenses.flatMap(_.travelCosts))(withAdjustmentValidations),
+        doValidationFor("/foreignProperty/expenses/costOfServices", foreignProperty.expenses.flatMap(_.costOfServices))(withAdjustmentValidations),
+        doValidationFor("/foreignProperty/expenses/residentialFinancialCost", foreignProperty.expenses.flatMap(_.residentialFinancialCost))(withAdjustmentValidations),
+        doValidationFor("/foreignProperty/expenses/other", foreignProperty.expenses.flatMap(_.other))(withAdjustmentValidations),
+        doValidationFor("/foreignProperty/expenses/consolidatedExpenses", foreignProperty.expenses.flatMap(_.consolidatedExpenses))(withAdjustmentValidations)
+      ).flatten
+    }
+
+    def validateForeignFhlEea(foreignFhlEea: FhlEea): List[MtdError] = {
+      List(
+        doValidationFor("/foreignFhlEea/income/rentIncome", foreignFhlEea.income.flatMap(_.rentIncome))(withAdjustmentValidations),
+        doValidationFor("/foreignFhlEea/expenses/premisesRunningCosts", foreignFhlEea.expenses.flatMap(_.premisesRunningCosts))(withAdjustmentValidations),
+        doValidationFor("/foreignFhlEea/expenses/repairsAndMaintenance", foreignFhlEea.expenses.flatMap(_.repairsAndMaintenance))(withAdjustmentValidations),
+        doValidationFor("/foreignFhlEea/expenses/financialCosts", foreignFhlEea.expenses.flatMap(_.financialCosts))(withAdjustmentValidations),
+        doValidationFor("/foreignFhlEea/expenses/professionalFees", foreignFhlEea.expenses.flatMap(_.professionalFees))(withAdjustmentValidations),
+        doValidationFor("/foreignFhlEea/expenses/travelCosts", foreignFhlEea.expenses.flatMap(_.travelCosts))(withAdjustmentValidations),
+        doValidationFor("/foreignFhlEea/expenses/costOfServices", foreignFhlEea.expenses.flatMap(_.costOfServices))(withAdjustmentValidations),
+        doValidationFor("/foreignFhlEea/expenses/other", foreignFhlEea.expenses.flatMap(_.other))(withAdjustmentValidations),
+        doValidationFor("/foreignFhlEea/expenses/consolidatedExpenses", foreignFhlEea.expenses.flatMap(_.consolidatedExpenses))(withAdjustmentValidations)
+      ).flatten
+    }
+
+    List(flattenErrors(List(
+      model.foreignFhlEea.map(validateForeignFhlEea).getOrElse(NoValidationErrors),
+      model.foreignProperty.map(validateForeignProperty).getOrElse(NoValidationErrors)
+    )))
```

Comment:
On the UK Property validator, we have a validation in place so that you can’t supply both FHL and non-FHL. Is this validation not required for this endpoint? Can you double check with Francis? If we need that kind of validation, something like this will do (along with a corresponding test):

```suggestion
    List(flattenErrors(List(
      (model.foreignFhlEea, model.foreignProperty) match {
        case (Some(foreignFhlEea), None) => validateForeignFhlEea(foreignFhlEea)
        case (None, Some(foreignProperty)) => validateForeignProperty(foreignProperty)
        case _ => List(RuleIncorrectOrEmptyBodyError)
      }
    )))
```

Obviously if you can think of a better way to do it then by all means... 👀